app/db/queries.py

```diff
@@ -5,44 +5,74 @@
 
 
 def get_messages_that_contain_word(word: str) -> List[Message]:
-    pass
+    return list(Message.objects.select_related().filter(
+        text__icontains=word
+    ))
 
 
 def get_untitled_chats() -> List[Chat]:
-    pass
+    return list(Chat.objects.filter(title__icontains="Untitled"))
 
 
 def get_users_who_sent_messages_in_2015() -> List[str]:
-    pass
+    return [
+        (user.first_name, user.last_name)
+        for user in User.objects.filter(message__sent__year=2015)
+    ]
 
 
 def get_actual_chats() -> List[Chat]:
-    pass
+    return Chat.objects.filter(message__sent__year__gt=2020)
 
 
 def get_messages_contain_authors_first_name():
-    pass
+    return Message.objects.select_related().filter(
+        text__contains=F('user__first_name')
+    )
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> List[User]:
-    pass
+    return User.objects.filter(Q(message__text__startswith='a')
+                               | Q(message__text__startswith='m'))
 
 
 def get_delivered_or_admin_messages() -> List[Message]:
-    pass
+    return [
+        message
+        for message in Message.objects.select_related().filter(
+            Q(user__username__startswith='admin') | Q(is_delivered=True))
+    ]
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return Message.objects.filter(
+        user__first_name=first_name,
+    ).count()
 
 
 def get_top_users_by_number_of_the_messages() -> List[User]:
-    pass
+    return User.objects.annotate(
+        num_messages=Count('message')
+    ).order_by('-num_messages')[:3]
 
 
 def get_last_5_messages_dicts() -> List[dict]:
-    pass
+    return [
+        {
+            'text': message.text,
+            'from': message.user.username
+        }
+        for message in Message.objects.prefetch_related(
```

Comment:
It would be more relevant to use select_related here (prefetch_related is used for many to many relations)

Reply:
And why did you approve the solution in this case?